.github/workflows/pytest.yml

```diff
@@ -0,0 +1,29 @@
+name: Run Pytest on Pull Request
+
+on:
+  pull_request:
+    branches:
+      - main  # Change this to your default branch if it's not 'main'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
```

Comment:
the action versions are outdated. See https://github.com/Akkudoktor-EOS/EOS/blob/156eddc6ef878faae84ceba7b0997a269e64d197/.github/workflows/pre-commit.yml#L13